app/decorators/models/solidus_bolt/shipment_decorator.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module SolidusBolt
+  module ShipmentDecorator
+    def self.prepended(base)
+      base.class_eval do
+        before_save :update_bolt_tracking_info
+      end
+    end
+
+    def update_bolt_tracking_info
+      return unless tracking_was.nil?
+      return unless tracking_changed?
+
+      payment = order&.payments&.completed&.last
+      return unless payment&.payment_method.instance_of?(SolidusBolt::PaymentMethod)
+
+      transaction_reference = payment&.response_code
+      return if transaction_reference.blank?
+
+      SolidusBolt::ShipmentTrackingJob.perform_later(transaction_reference: transaction_reference, shipment: self)
```

Comment:
What will happen if the order contains more than one shipment?

Reply:
I'm not sure yet if Bolt supports multiple shipments like Solidus.

Nothing in the bolt docs points to supporting multiple shipments and also this API returns an error if we have already added one instance of tracking details to the Order on Bolt.

Reply:
Sure